src/Neo/SmartContract/ApplicationEngine.cs

```diff
@@ -65,8 +65,11 @@ public partial class ApplicationEngine : ExecutionEngine
 
         private static Dictionary<uint, InteropDescriptor> services;
         // Total amount of GAS spent to execute.
-        // In the unit of datoshi, 1 datoshi = 1e-8 GAS, 1 GAS = 1e8 datoshi
-        private readonly long _feeAmount;
+        // In the unit of datoshi with two decimals, 1 datoshi = 1e-10 GAS, 1 GAS = 1e10 datoshi
```

Comment:
Technically, 1 datoshi is still 1e-8 GAS since we don't change GAS decimals. Ref. https://github.com/neo-project/proposals/blob/master/nep-28.mediawiki#specification.